### PR TITLE
fix(swap): display input amount usd value even if there is no quote

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

In case when we haven't get a quote for any reason (most frequent reason is "Fees exceed from amount") we don't display a USD amount for input amount.
But we should display it because it will help a user to understand what the difference between input amount and fee.

image

  # To Test

1. Setup a trade with "Fees exceed from amount" state (Sell 2 COW to WETH on Mainnet)
- [ ] AR: USD value for input amount is not displayed
- [ ] ER: USD value for input amount is displayed

  # Background

`useTradeUsdAmounts` hook is intended for cases when a trade is ready (we have a quote). Since in the swap form we have a different case we should use just `useUsdAmount` hook

### CODE REVIEW

**Review by alfetopito — APPROVED** [review]

**Comment by elena-zh** [comment]
Hey @shoom3301 , great, but some nitpicking: 
1. (it is on Prod): what about Buy orders? There is no USD estimation when I fill in the only one token into the Buy field+amount. Besides, eternal loader is displayed there
image


2. Should we extend this behavior on Limit/TWAP pages? image
image

**Comment by shoom3301** [comment]
@elena-zh thanks! Fixed both

**Comment by shoom3301** [comment]
@elena-zh could you please check price impact warning?
It was flickering some time ago, I've touched the code related to this, just wanna make sure it doesn't flicker again

File: apps/cowswap-frontend/src/modules/usdAmount/hooks/useTradeUsdAmounts.ts [hunk 1]
```diff
@@ -25,10 +25,12 @@ export function useTradeUsdAmounts(
...
   outputAmount: Nullish<CurrencyAmount<Currency>>,
   inputCurrency?: Nullish<TokenWithLogo>,
-  outputCurrency?: Nullish<TokenWithLogo>
+  outputCurrency?: Nullish<TokenWithLogo>,
+  dontWaitBothAmounts?: boolean
 ): TradeUSDAmounts {
   const isWrapOrUnwrap = useIsWrapOrUnwrap()
-  const isTradeReady = !isWrapOrUnwrap && !isFractionFalsy(inputAmount) && !isFractionFalsy(outputAmount)
+  const areAmountsReady = !isFractionFalsy(inputAmount) && !isFractionFalsy(outputAmount)
+  const isTradeReady = !isWrapOrUnwrap && (dontWaitBothAmounts ? true : areAmountsReady)
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/usdAmount/hooks/useTradeUsdAmounts.ts:33]: How about:

```suggestion
  const isTradeReady = !isWrapOrUnwrap && (dontWaitBothAmounts || areAmountsReady)
```

**Review by elena-zh — APPROVED** [review]
looks good now, thanks!

**Pull request merged**